### PR TITLE
Update Golang dependencies to Knative 0.26

### PR DESCRIPTION
**Pull request by markusthoemmes**
This brings a sleeeeeeew of stuff, most notably K8s 1.21 deps. Let's shake the tree.

Also in this:
- Patches for the route retrying to resemble what we're doing over in Serving.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Depends on https://github.com/openshift-knative/serverless-operator/pull/1274

/hold

File: go.mod [hunk 1]
```diff
@@ -15,35 +15,32 @@ require (
...
-	k8s.io/api v0.20.7
-	k8s.io/apimachinery v0.20.7
-	k8s.io/client-go v12.0.0+incompatible
-	knative.dev/eventing v0.25.1
+	go.uber.org/zap v1.19.1
+	k8s.io/api v0.21.4
+	k8s.io/apimachinery v0.21.4
+	k8s.io/client-go v0.21.4
+	knative.dev/eventing v0.26.1
 	knative.dev/eventing-kafka v0.0.0-00010101000000-000000000000
```
Comment (matzew) [inline go.mod:25]: that version string always gets my attention :exploding_head:

File: hack/patches/003-routeretry.patch [hunk 2]
```diff
@@ -0,0 +1,51 @@
+diff --git a/vendor/knative.dev/pkg/test/spoof/openshift_checks.go b/vendor/knative.dev/pkg/test/spoof/openshift_checks.go
```
Comment (markusthoemmes) [inline hack/patches/003-routeretry.patch:1]: This patch is required to make up for the dropped `EventuallyMatchesBody`, which was essentially a universal retry.

**Pull request merged**